include/maidsafe/common/rsa.h

```diff
@@ -85,10 +86,15 @@ PlainText Decrypt(const CipherText& data, const PrivateKey& private_key);
 
 Signature Sign(const PlainText& data, const PrivateKey& private_key);
 
+std::vector<byte> Sign(const std::vector<byte>& data, const PrivateKey& private_key);
```

Comment:
The purpose was to allow easy serialisation (to array), so we could have another type that is Signature that is basically based on vector<bytes> as a tagged value with a slightly different name.  It is no real biggie, but the Signature type is based on strings rather than vector<byte> If you feel very strongly then we can make it Signature type, but feels a bit strange to have the mix   of sting and vector<byte> if you see what I mean. I agree it would be better as a type though.